Automattic/wp-calypso
# Sites Management Page: Include `?v=<rounded-updated-at>` to bust mShot

See conversation in https://github.com/Automattic/wp-calypso/issues/66524#issuecomment-1213372612

## Proposed Changes

Adds a `?v=<site-updated-at>` (rounded down to the nearest hour) to the `mShotsUrl` to semi-regularly refresh the screenshot when a site is updated.

I'm not sure if this is a "blessed" use of mShots. Notably, if the site's full page cache doesn't discard the `v` query param, this approach will likely bypass the full page cache, and take a longer time to generate a mShot.

## Testing Instructions

1. Navigate to `/sites-dashboard`.
2. Verify the `v` query parameter on a mShots URL is rounded down to the nearest start of the hour (e.g. `https://s0.wp.com/mshots/v1/https%3A%2F%2Fdanielwoatest.wpcomstaging.com%3Fv%3D2022-08-12T17%253A00%253A00.000Z?vpw=1200&vph=1200&w=748&countToRefresh=67`).
3. Comment out the `updatedAt.setMinutes( 0 );` and `updatedAt.setSeconds( 0 );` calls in this pull request.
4. Make a change to one of your sites.
5. Navigate back to the Sites Management Page and see the screenshot refresh once your sites data is refreshed.
